### PR TITLE
Proposal Page: Add Proposal Context

### PR DESCRIPTION
**Pull request by pavloMandryk**
It displays Proposal context and Challenger data on the Proposal Page, so you don't have to click 'Accept settlement' to read it.

This is the [figma](https://www.figma.com/file/IGMjM4UyW2yp9ImEUsHduG/%E2%9C%A8Improvements-based-on-UX-research?node-id=0%3A1) used.

This solves [issue #268](https://github.com/1Hive/gardens/issues/268).

### CODE REVIEW

File: src/components/Garden/ProposalDetail/ProposalDetail.js [hunk 1]
```diff
@@ -388,6 +416,148 @@ function ProposalDetail({
...
+      setChallengerAddress(challengeData.challenger)
+    }
+    getChallengeData()
+
+    async function fetchChallengerProfile() {
+      const challengerProfile = await getProfileForAccount(challengerAddress)
+      setChallengerData(challengerProfile)
+    }
+    fetchChallengerProfile()
+  }, [challengerAddress])
```
Comment (fabriziovigevani) [inline src/components/Garden/ProposalDetail/ProposalDetail.js:444]: What do you think of creating a hook `useChallenge` or similar which encapsulates this logic.

Also we could have one state variable object and set all values in one go.
```suggestion
  const [challenge, setChallenge] = useState()
  const [showArgument, setShowArgument] = useState(false)

  const { agreementActions } = useActions()

  useEffect(() => {
    async function getChallengeData() {
      const challengeData = await agreementActions.getChallenge(
        proposal.challengeId
      )
      const challengerProfile = await getProfileForAccount(
        challengeData.challenger
      )

      setChallenge({
        context: hexToUtf8(challengeData.context),
        challenger: { address: challengeData.challenger, ...challengerProfile },
      })
    }
    getChallengeData()
  }, [agreementActions, proposal.challengeId])
```

File: src/components/Garden/ProposalDetail/ProposalDetail.js [hunk 2]
```diff
@@ -388,6 +416,148 @@ function ProposalDetail({
...
+      >
+        <div
+          css={`
+            display: grid;
+            grid-template-columns: ${5 * GU}px ${35 * GU}px;
+            align-items: center;
+          `}
+        >
+          <img src={warningIcon} />
+          <h1>Your proposal has been challenged</h1>
```
Comment (fabriziovigevani) [inline src/components/Garden/ProposalDetail/ProposalDetail.js:464]: We should check that the connected account is the author of the proposal

**Review by fabriziovigevani — COMMENTED** [review]
Amazing job! 

Did a partial review, will do a second one after.

Let me know what do you think.

File: src/components/Garden/ProposalDetail/ProposalDetail.js [hunk 3]
```diff
@@ -143,161 +154,178 @@ function ProposalDetail({
...
+                  <ProposalActions
+                    proposal={proposal}
+                    onChangeSupport={() => handleShowModal('update')}
+                    onExecuteProposal={() => handleShowModal('execute')}
+                    onRequestSupportProposal={() => handleShowModal('support')}
                   />
+                </section>
+              </Box>
+              {(statusData.challenged || statusData.settled) &&
+                connectedAccount && (
```
[the remark this answers is not in the document]
Comment (pavloMandryk) [reply src/components/Garden/ProposalDetail/ProposalDetail.js:317]: In order to get challengeData we are using the useContract() hook on agreementContract. The useContract() hook requires a provider so, as I understand, it won't be able to reach for challengeData if there is no account connected. Did I get it right?

File: src/hooks/useChallenge.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+      setChallenge({
+        context: hexToUtf8(challengeData.context),
+        challenger: { address: challengeData.challenger, ...challengerProfile },
+      })
+      setLoading(false)
+    }
+
+    getChallengeData()
+  }, [proposal.challengeId])
```
Comment (fabriziovigevani) [inline src/hooks/useChallenge.js:32]:
```suggestion
  }, [agreementActions, proposal])
```

File: src/hooks/useChallenge.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    }
+    async function getChallengeData() {
+      const challengeData = await agreementActions.getChallenge(
+        proposal.challengeId
+      )
+      const challengerProfile = await getProfileForAccount(
+        challengeData.challenger
+      )
+
+      setChallenge({
```
Comment (fabriziovigevani) [inline src/hooks/useChallenge.js:24]: Let´s use the `useMounted` hook and update the state only if `mounted()`

File: src/components/Garden/ProposalDetail/ProposalDetail.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (fabriziovigevani) [reply src/components/Garden/ProposalDetail/ProposalDetail.js:315]: Ah yeah we should update that. We should use `useContractReadOnly` for this case.

Let´s remove the `getChallenge` function from `useActions`  and instead let´s update the `useChallenge` hook to use the hooks that i mentioned first: 

```
  const { connectedAgreementApp } = useGardenState()
  const agreementContract = useContractReadOnly(
    connectedAgreementApp.address,
    agreementAbi
  )

  const [challenge, setChallenge] = useState()
  const [loading, setLoading] = useState(true)

  useEffect(() => {
    if (!proposal) {
      return
    }
    async function getChallengeData() {
      const challengeData = await agreementContract.getChallenge(
        proposal.challengeId
      )
      const challengerProfile = await getProfileForAccount(
        challengeData.challenger
      )

      setChallenge({
        context: hexToUtf8(challengeData.context),
        challenger: { address: challengeData.challenger, ...challengerProfile },
      })
      setLoading(false)
    }

    getChallengeData()
  }, [agreementContract, proposal])

  return { challenge, loading }
```

File: src/hooks/useChallenge.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            address: challengeData.challenger,
+            ...challengerProfile,
+          },
+        })
+      }
+      setLoading(false)
+    }
+
+    getChallengeData()
+  }, [proposal.challengeId, mounted, agreementContract])
```
Comment (fabriziovigevani) [inline src/hooks/useChallenge.js:46]: We need to add the entire proposal as dependency.
```suggestion
  }, [agreementContract, mounted, proposal])
```

File: src/hooks/useChallenge.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react'
+
+import { useContractReadOnly } from './useContract'
+import { useMounted } from './useMounted'
+import { useGardenState } from '@providers/GardenState'
```
Comment (fabriziovigevani) [inline src/hooks/useChallenge.js:5]:
```suggestion
import { useContractReadOnly } from './useContract'
import { useGardenState } from '@providers/GardenState'
import { useMounted } from './useMounted'
```

**Review by fabriziovigevani — APPROVED** [review]
Just a few small comments and let´s merge it 🙌

**Pull request merged**